eventstore/model.go

```diff
@@ -136,38 +132,6 @@ func (m *Model) FindJSON(q JSONQuery) (ret []string, err error) {
 	return
 }
 
-// Reduce processes an event into the model.
-func (m *Model) Reduce(event core.Event) error {
```

Comment:
Yep. Models aren't reducers anymore.
TBH, this always felt weird. They kind of checked if the current event was for them (with that `if event.Type() != m.name` code), and if it corresponds, call the event codec.

As we can see, being in the Model world to do this adds little value. Most of the code in this old method does `m.store.XXXX` for things. Which was already a smell.

Now that what is reduce are aggregates of things of a txn, this becomes even more apparent. In the future, a txn might change instances of multiple models. A single model can't reduce really txn changes.

Now the `Store` itself is the reducer. (The only reducer now in the code). That makes more sense. A store will receive a slice of Events (which might contain changes for multiple models), and will take care of applying the changes.

Reply:
Cool, I like it... conceptually simpler